clients/src/main/java/org/apache/kafka/clients/consumer/internals/PartitionAssignor.java

```diff
@@ -73,67 +78,171 @@ default void onAssignment(Assignment assignment, int generation) {
         onAssignment(assignment);
     }
 
+    /**
+     * Indicate which rebalance protocol this assignor can would work with;
+     * By default it should always work with {@link RebalanceProtocol#EAGER}.
+     */
+    default List<RebalanceProtocol> supportedProtocols() {
+        return Collections.singletonList(RebalanceProtocol.EAGER);
+    }
+
+    /**
+     * Return the version of the assignor which indicate how the user metadata encodings
```

Comment:
nit: indicate -> indicates?

Do you think it's worth mentioning something about how the coordinator uses this?